app/controllers/fhir/r4/api_controller.rb

```diff
@@ -424,6 +424,120 @@ def all
+    if client_app.exported_recently?
+      status_too_many_requests_with_custom_errors(['Client already initiated an export of this type in the last 15 minutes. Please try again later']) && return
+    end
+
+    begin
+      since = params.permit(:_since)[:_since]
+      since = DateTime.parse(since) unless since.nil?
+    rescue Date::Error
+      status_unprocessable_entity_with_custom_errors(['Invalid Date in _since parameter. Please use the FHIR instant datatype'], '') && return
+    end
+
+    # Create the download to uniquely id this request, and queue the ExportFhirJob
+    download = ApiDownload.create(application_id: client_app.id, url: request.url)
+    export_job = ExportFhirJob.perform_later(client_app, download, { since: since })
+    # Add the job_id of the job to the download for reference in status requests
+    download.update(job_id: export_job.provider_job_id)
+
+    respond_to do |format|
+      format.any { head(:accepted, content_location: "#{root_url}fhir/r4/ExportStatus/#{download.id}") }
+    end
+  end
+
+  # Give an update on the status of a bulk FHIR data request
+  #
+  # GET /fhir/r4/ExportStatus/[:id]
+  def export_status
+    id = params.permit(:id)[:id]
+    download = ApiDownload.find_by_id(id)
+
+    case Sidekiq::Status.status(download&.job_id)
+    when :complete
+      response_json = {
+        transactionTime: Sidekiq::Status.get(download.job_id, :transaction_time),
+        request: download.url,
+        requiresAccessToken: true,
+        output: ActiveStorage::Blob.where(id: download.files.pluck(:blob_id)).pluck(:filename).map do |file|
+          {
+            type: file.split('.').first,
+            url: "#{root_url}fhir/r4/ExportFiles/#{id}/#{file}"
+          }
+        end
+      }
+      render json: response_json.to_json, status: :ok
+    when :failed
+      status_server_error(['Export failed'])
+    when :working, :queued, :retrying
+      head :accepted
+    else
+      status_not_found_with_custom_errors(['No export found for this ID'])
+    end
+  end
+
+  # Return an bulk exported ndjson file
+  #
+  # GET /ExportFiles/[:id]/[:filename]
+  def export_files
+    id = params.permit(:id)[:id]
+    filename = params.permit(:filename)[:filename]
+    resource_type = filename&.split('.')&.first&.downcase
+
+    case resource_type
+    when 'patient'
+      return if doorkeeper_authorize!(*PATIENT_READ_SCOPES)
+    when 'observation'
+      return if doorkeeper_authorize!(*OBSERVATION_READ_SCOPES)
+    when 'questionnaireresponse'
+      return if doorkeeper_authorize!(*QUESTIONNAIRE_RESPONSE_READ_SCOPES)
+    when 'relatedperson'
+      return if doorkeeper_authorize!(*RELATED_PERSON_READ_SCOPES)
+    when 'immunization'
+      return if doorkeeper_authorize!(*IMMUNIZATION_READ_SCOPES)
+    when 'provenance'
+      return if doorkeeper_authorize!(*PROVENANCE_READ_SCOPES)
+    else
+      status_not_found_with_custom_errors(["Invalid ResourceType '#{resource_type}'"]) && return
+    end
```

Comment:
Yeah if I break this out it should be easy enough to support `_type`, so I will do that. Another thing we could do is support returning only partial data (even if `_type` is not set). So if someone only has permission to read `Patient`, they will only get back `Patient`s, and not get an error for not having all the scopes. Personally I think for now I want to avoid this, and make someone specify `_type` if they want fewer resources, keeping it explicit. But I just wanted to raise the idea in case you disagree.

Reply:
> Personally I think for now I want to avoid this, and make someone specify _type if they want fewer resources, keeping it explicit. 
Just so I'm clear. `_type.nil? # true` get everything. `_type # ['Patient', 'Observation']` get patient and observation only? Seems good.

Reply:
Yeah, you got it, and more explicitly, if you don't specify `_type`, but you only have the scopes required to read `Patient`, that is an error. The spec is a little vague, and in that case we probably could just return `Patients` even though `_type` didn't specifically specify `Patient`, but I don't love that idea.

Reply:
I've gone back and forth on this, and I've ended up changing my mind on if we should support `_type` yet. I think in the interest of simplicity, it's best not to include `_type` until we are asked for it. It'll keep the code simpler, and it makes things simpler to understand for the API users (who we know at this point aren't interested in `_type`). I still will break this out in another method though, that way it'll be a little easier if we do have demand for `_type` in the future. Does that sound fair to you?

Reply:
Added error message for `_type` and `_outputFormat` [here](https://github.com/SaraAlert/SaraAlert/pull/987/commits/6977cd314b84a7c5ecfd8d2d2ac200bf5d218e7b#diff-e79e38176ee697a8541cffba23c60d9decaa69382ca5c6e875da033508e5b428R453-R454), but let me know if you feel really strongly that we should support `_type` now.

Reply:
I'm good with this approach.